packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp

```diff
@@ -383,17 +351,18 @@ void DistributorActor::doPosts(const DistributorPlan& plan,
     typedef typename ExpView::array_layout Layout;
     typedef typename ExpView::device_type Device;
     typedef typename ExpView::memory_traits Mem;
+
+    // This buffer is long enough for only one message at a time.
+    // Thus, we use DISTRIBUTOR_SEND always in this case, regardless
+    // of sendType requested by user. 
+    // This code path formerly errored out with message:
+    //     Tpetra::Distributor::doPosts(3 args, Kokkos): 
+    //     The "send buffer" code path
+    //     doesn't currently work with nonblocking sends.
+    // Now, we opt to just do the communication in a way that works.
```

Comment:
I think a print statement that's only active in debug makes sense in this case.

For Rsend/Ssend, there is some protection, because a DEPRECATED_CODE=OFF build will give a compiler warning if they try to use those enum values.  I wouldn't mind a warning that we are converting Rsend/Ssend to Send printed always (release or debug), but only if DEPRECATED_CODE=ON.

Reply:
I'll add the print statements.

Reply:
I'd argue that your new printout is sufficiently descriptive that it's better to delete the comment since we don't really need it anymore.  But I won't hold up the PR over this.